src/Graph/Trees/Heap.php

```diff
@@ -0,0 +1,179 @@
+<?php
+
+namespace Rubix\ML\Graph\Trees;
+
+use InvalidArgumentException;
+
+/**
+ * Heap
+ *
+ * A heap is a specialized tree-based data structure that satisfies the heap property:
+ * in a max heap, for any given node C, if P is a parent node of C, then the key of P
+ * is greater than or equal to the key of C. In a min heap, the key of P is less than or equal to the key of C.
+ *
+ * References:
+ * [1] Black (ed.), Paul E. (2004-12-14)
+ * <https://xlinux.nist.gov/dads/HTML/heap.html>
+ *
+ * @category    Machine Learning
+ * @package     RubixML
+ * @author      Rich Davis
+ */
+class Heap
+{
+    /**
+     * An array of words from a corpus arranged in a manner that maintains the heap invariant.
```

Comment:
Are you referring to the heap property?

The heap is well, the heap. It's the specific object the class creates. 

There is no min or max heap invariant, here is the definition: https://docs.python.org/3/library/heapq.html

Reply:
I am referring to the name you chose to give to this variable ... Could we name is something other than heap since we already called the aggregate root a "heap".

Reply:
I can't see which variable you're referring to in the snippet for this comment. It just shows the Heap class.